StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/MaintainabilityRules/SA1653CodeFixProvider.cs

```diff
@@ -0,0 +1,129 @@
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(SA1653UseTrailingCommasInMultiLineInitializers.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        MaintainabilityResources.SA1653CodeFix,
+                        cancellationToken => GetTransformedDocumentAsync(context.Document, diagnostic, cancellationToken),
+                        nameof(SA1653CodeFixProvider)),
+                    diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var parent = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).Parent;
+
+            SyntaxNode newParent = null;
+            switch (parent.Kind())
+            {
+                case SyntaxKind.ObjectInitializerExpression:
+                case SyntaxKind.ArrayInitializerExpression:
+                case SyntaxKind.CollectionInitializerExpression:
+                    newParent = RewriteInitializer((InitializerExpressionSyntax)parent);
+                    break;
+
+                case SyntaxKind.AnonymousObjectCreationExpression:
+                    newParent = RewriteAnonymousObjectInitializer((AnonymousObjectCreationExpressionSyntax)parent);
+                    break;
+
+                default:
+                    throw new InvalidOperationException("Unknown initializer type: " + parent.Kind());
+            }
+
+            var newSyntaxRoot = syntaxRoot.ReplaceNode(parent, newParent);
+
+            var newDocument = document.WithSyntaxRoot(newSyntaxRoot.WithoutFormatting());
+            return newDocument;
+        }
+
+        private static SyntaxNode RewriteInitializer(InitializerExpressionSyntax initializer)
+        {
+            var existingItems = new List<ExpressionSyntax>(initializer.Expressions);
+            var last = existingItems.Last();
+            existingItems.Remove(last);
+            existingItems.Add(last.WithTrailingTrivia(last.GetTrailingTrivia().WithoutTrailingWhitespace()));
+
+            var existingSeparators = initializer.Expressions.GetSeparators();
+            var newSeparators = new List<SyntaxToken>(existingSeparators);
+            newSeparators.Add(SyntaxFactory.Token(SyntaxKind.CommaToken).WithTrailingTrivia(last.GetTrailingTrivia()));
```

Comment:
:bug: This statement will duplicate any existing trailing trivia for the last item. The trailing trivia should be stripped from the last item and added after the separator probably.

Example:
```csharp
var x = new[]
{
  1,
  2,
  3 /* last item */
}
```